projects/plugins/jetpack/extensions/blocks/subscriptions/hooks/use-newsletter-categories.js

```diff
@@ -0,0 +1,35 @@
+import apiFetch from '@wordpress/api-fetch';
+import { select } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { CONNECTION_STORE_ID } from '../../../../../../js-packages/connection';
+
+const useNewsletterCategories = () => {
+	const [ data, setData ] = useState( [] );
+	const [ enabled, setEnabled ] = useState( false );
+	const [ error, setError ] = useState( false );
+	const [ loading, setLoading ] = useState( true );
+
+	useEffect( () => {
+		const blogId = select( CONNECTION_STORE_ID ).getBlogId();
+
+		const fetchData = async () => {
+			try {
+				const newsLetterCategories = await apiFetch( {
+					url: `https://public-api.wordpress.com/wpcom/v2/sites/${ blogId }/newsletter-categories`,
```

Comment:
We've typically not queried `public-api` directly in our blocks. Instead we query a local endpoint that in turn queries WordPress.com (or makes a direct request when we're on WordPress.com).  The local endpoints live in `projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/`. 

There is some documentation about it here:

https://github.com/Automattic/jetpack/blob/695768a21b1341684a4da54fa4abc39093cd7ed8/projects/plugins/jetpack/extensions/README.md#L262-L271

Reply:
Thanks for the pointers once again @jeherve .

I've made the needed API changes: https://github.com/Automattic/jetpack/pull/32666/commits/596bcef21f4fc3f89fc44768f728e72f86cede54

I do have a question regarding defusion: When I test with the jetpack-downloader, I can see the endpoint file on my WP.COM Simple site. However, even after deleting the existing endpoint, I cannot get it to execute it using the WP.com console. Do files inside `wpcom-endpoints` not get loaded on Simple Sites?

It's not a problem for now, since everything works. Just interested.

Reply:
> Do files inside wpcom-endpoints not get loaded on Simple Sites?

Yeah, it's a bit confusing. You'd need to do something like D110282-code (check the last file being added, or other files in that directory for some more examples).

Reply:
Thank you! It's much cleared now. I'll create a separate patch for WPCOM, removing the endpoint, after the Jetpack release happens on WPCOM.